src/lib/components/EventRenderLevelLimit.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import { levelsToRender } from "$lib/state";
+  import { createEventDispatcher } from "svelte";
+
+  let inputValue = $levelsToRender;
+
+  const dispatch = createEventDispatcher<{
+    update: { limit: number };
+  }>();
```

Comment:
Svelte 5 [recommends](https://svelte.dev/docs/svelte/v5-migration-guide#Event-changes-Component-events) setting up events as callback functions passed to the component as props, rather than using the `createEventDispatcher` API.